src/Symfony/Component/Cache/Traits/PdoTrait.php

```diff
@@ -316,6 +322,11 @@ protected function doSave(array $values, $lifetime)
                 $this->createTable();
             }
             $stmt = $conn->prepare($sql);
+        } catch (\PDOException $e) {
+            if (\in_array($this->driver, ['pgsql', 'sqlite', 'sqlsrv'], true)) {
```

Comment:
`if (!$conn->inTransaction() || \in_array($this->driver, ['pgsql', 'sqlite', 'sqlsrv'], true)) {`
(same below)